diff --git a/crates/c-api/CMakeLists.txt b/crates/c-api/CMakeLists.txt
index ff92a091e..07c667a77 100644
--- a/crates/c-api/CMakeLists.txt
+++ b/crates/c-api/CMakeLists.txt
@@ -61,19 +61,27 @@ endif()
 if(ANDROID)
 	# Rust attempts to use libgcc.a on NDK versions r23-beta3 and up
 	# but it has been replaced with libunwind.a (rust-lang/rust#85806)
-	file(WRITE ${CMAKE_BINARY_DIR}/libgcc.a "INPUT(-lunwind)")
+	file(WRITE ${CMAKE_BINARY_DIR}/libgcc.a "")
+	file(WRITE ${CMAKE_BINARY_DIR}/libunwind.a "")
 	# The version of the clang compiler is part of the libunwind.a path
 	file(STRINGS ${ANDROID_TOOLCHAIN_ROOT}/AndroidVersion.txt CLANG_VERSION_FILE)
 	list(GET CLANG_VERSION_FILE 0 CLANG_VERSION)
 
 	# Some crates use the compiler directly, environment variables
 	# are set to make them use the Android compiler
+	if(EXISTS ${ANDROID_TOOLCHAIN_ROOT}/bin/ld)
+		set(RUST_LINKER_PATH ${ANDROID_TOOLCHAIN_ROOT}/bin/ld)
+	elseif(EXISTS ${ANDROID_TOOLCHAIN_ROOT}/bin/ld.lld)
+		set(RUST_LINKER_PATH ${ANDROID_TOOLCHAIN_ROOT}/bin/ld.lld)
+	else()
+		message(FATAL_ERROR "Linker not found at ${ANDROID_TOOLCHAIN_ROOT}/bin")
+	endif()
 	set(WASMTIME_PREBUILD_COMMAND ${CMAKE_COMMAND} -E env
 	CC=${ANDROID_TOOLCHAIN_ROOT}/bin/clang
 	AR=${ANDROID_TOOLCHAIN_ROOT}/bin/llvm-ar
 	"RUSTFLAGS=-L ${CMAKE_SYSROOT}/usr/lib/${ANDROID_TARGET}/${ANDROID_NATIVE_API_LEVEL} \
 		-L ${ANDROID_TOOLCHAIN_ROOT}/lib64/clang/${CLANG_VERSION}/lib/linux/${ANDROID_ARCH_SHORT} \
-		-L ${CMAKE_BINARY_DIR} -C linker=${ANDROID_TOOLCHAIN_ROOT}/bin/ld")
+		-L ${CMAKE_BINARY_DIR} -C linker=${RUST_LINKER_PATH}")
 endif()
 include(ExternalProject)
 find_program(WASMTIME_CARGO_BINARY cargo)
