diff --git a/source/m3_core.h b/source/m3_core.h
index c470c3b..964912a 100644
--- a/source/m3_core.h
+++ b/source/m3_core.h
@@ -170,7 +170,7 @@ M3CodePageHeader;
 #define d_m3MaxSaneElementSegments          10000000
 #define d_m3MaxSaneDataSegments             100000
 #define d_m3MaxSaneTableSize                10000000
-#define d_m3MaxSaneUtf8Length               10000
+#define d_m3MaxSaneUtf8Length               10000000
 #define d_m3MaxSaneFunctionArgRetCount      1000    // still insane, but whatever
 
 #define d_externalKind_function             0
